specs/DataSet.rst

```diff
@@ -0,0 +1,253 @@
+    The description can be a simple string or a string-to-string dictionary.
+    If optional is provided and is true, then the parameter is optional in each result.
+
+DataSet
+~~~~~~~
+
+DataSet()
+    Creates a DataSet with no parameters.
+
+DataSet(specs)
+    Creates a DataSet for the provided list of parameter specifications.
+    Each item in the list should either be a QCoDeS Parameter, a tuple of a Parameter and a Boolean, or a ParamSpec object.
+    A Parameter or a Parameter tupled with a false value indicates a required parameter; a Parameter tupled with a true value indicates an optional parameter.
+
+DataSet.add_parameter(spec)
+    Adds a parameter to an existing DataSet.
+    The spec should either be a QCoDeS Parameter, a tuple of a Parameter and a Boolean, or a ParamSpec object.
+    A Parameter or a Parameter tupled with a false value indicates a required parameter; a Parameter tupled with a true value indicates an optional parameter.
+    It is an error to add a parameter to a non-empty DataSet.
+
+DataSet.add_parameters(specs)
+    Adds a list of parameters to an existing DataSet.
+    Each item in the list should either be a QCoDeS Parameter, a tuple of a Parameter and a Boolean, or a ParamSpec object.
+    A Parameter or a Parameter tupled with a false value indicates a required parameter; a Parameter tupled with a true value indicates an optional parameter.
+    It is an error to add a parameter to a non-empty DataSet.
+
+DataSet.add_metadata(tag=, info=)
+    Adds metadata to the current DataSet.
+    The metadata is stored under the provided tag.
+    It is an error to add metadata to a completed DataSet.
+
+Writing
+-------
+
+DataSet.add_result(**kwargs)
+    Adds a result to the DataSet.
+    Keyword parameters should have the name of a parameter as the keyword and the value to associate as the value.
+    If there is only one positional parameter and it is a dictionary, then it is interpreted as a map from parameter name to parameter value.
+    It is an error for a value for the same parameter to be specified both using a positional parameter or dictionary parameter and using a keyword,
+    It is an error to provide a value for a key or keyword that is not the name of a parameter in this DataSet.
+    It is an error to add a result to a completed DataSet.
+
+DataSet.add_results(args)
+    Adds a sequence of results to the DataSet.
+    The single argument should be a sequence of dictionaries, where each dictionary provides the values for all of the parameters in that result.
+    See the add_result method for a description of such a dictionary.
+    The order of dictionaries in the sequence will be the same as the order in which they are added to the DataSet.
+    It is an error to add results to a completed DataSet.
+
+DataSet.complete()
+    Marks the DataSet as completed.
+
+Access
+------
+
+DataSet.length
+    This attribute holds the current number of results in the DataSet. 
```

Comment:
You measure single shots for a 2qubit tomography. 
Two rows are all the raw shots (1000s of them) 
You then add three other rows containing the correlations for each bin (depending on how you choose it 36*3 values).